packages/iov-bns/src/encode.ts

```diff
@@ -10,26 +11,55 @@ import {
   TransactionKind,
   UnsignedTransaction,
 } from "@iov/bcp-types";
+import { Algorithm, PrivateKeyBundle, PublicKeyBundle } from "@iov/tendermint-types";
 
 import * as codecImpl from "./codecimpl";
-import { encodeFullSig, encodeToken } from "./types";
+import { encodeFullSig } from "./types";
 import { decodeBnsAddress, keyToAddress, preimageIdentifier } from "./util";
 
-export const buildSignedTx = (tx: SignedTransaction): codecImpl.app.ITx => {
+export function encodePubkey(publicKey: PublicKeyBundle): codecImpl.crypto.IPublicKey {
```

Comment:
Bye, bye top-level lambdas.

I know, we agreed this needed to go

Reply:
There are some places in this PR where being forced to set return type adds real value to the code. This changes from _some interface_ to _the type we want to work with_

* https://github.com/iov-one/iov-core/pull/509/commits/6cdeee9f2fcadac7813602935e15fac9cac12ba4#diff-3a78df7aabe6b2e904eee5e9b98e2d19R46
* https://github.com/iov-one/iov-core/pull/509/commits/6cdeee9f2fcadac7813602935e15fac9cac12ba4#diff-3a78df7aabe6b2e904eee5e9b98e2d19R55